mlx/ops.cpp

```diff
@@ -2222,6 +2222,176 @@ array matmul(
   return out;
 }
 
+/** Computes: O = softmax(Q @ K.T) @ V; with parallelization across K-V sequence
+ * length **/
+array fast_inference_sdpa(
+    const array& queries,
+    const array& keys,
+    const array& values,
+    const float scale,
+    const std::optional<array>& mask,
+    StreamOrDevice s) {
+  auto q = queries;
+  auto k = keys;
+  auto v = values;
+  auto m = mask;
```

Comment:
I would just change the function names (and make them pass by values instead of ref to const) instead of doing the copy here.

Reply:
sorry, can you clarify?

I was mostly following that matmul() primitive in style:

```
array matmul(
    const array& in_a,
    const array& in_b,
    StreamOrDevice s /* = {} */) {
  auto a = in_a;
  auto b = in_b;
```